packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -453,8 +455,12 @@ export class GarbageCollector implements IGarbageCollector {
 		telemetryContext?: ITelemetryContext,
 	): Promise<IGCStats | undefined> {
 		const fullGC =
-			options.fullGC ??
-			(this.configs.runFullGC === true || this.summaryStateTracker.doesSummaryStateNeedReset);
+			this.fullGCModeForRecovery ||
+			(options.fullGC ??
+				(this.configs.runFullGC === true ||
+					this.summaryStateTracker.doesSummaryStateNeedReset));
+
+		this.fullGCModeForRecovery = false;
```

Comment:
You can't set it to false here because this GC may not succeed. You need to wait until the summary is acked before you can set it to false. Otherwise, if this GC run fails later or summary fails or summary is not acked, the next GC run will not be with fullGC.
The `GCSummaryStateTracker` does this for other state that needs to be tracked for a complete summary.

Reply:
Thanks for pointing this out! I moved this state to `GCSummaryStateTracker` and only clear it on summary ack.